andela/bp-esa-backend
# #165856629 Endpoint to handle update of partner slack details

#### What does this PR do?
Allow a user update partner slack channel details before retrying automation

#### Description of Task to be completed?
- install express-validator and yup for partner model/data validation
- implement handleValidation and validatePartner to validate request body
- implement updatePartnerSlackDetails controller action
- define routes for endpoint to update partner slack details

#### How should this be manually tested?
1. On terminal, run command: `npm install` to install new dependencies
2. Run command: `npm run start:dev`
3. On your postman, make a request to endpoint: `PUT /partners/:id` with request body
```
{
  general: { channelId: 'GJ6AKSWBC', channelName: 'p-accessmobile' },
  internal: {
    channelId: 'GJH7JJ6E8',
    channelName: 'p-accessmobile-int',
  },
}
```
where `id` in request params is the `partnerId` to update
4. Observe that a `200` response status code is returned with the updated partner details

#### Any background context you want to provide?
This is important because a slack automation could fail due to no partner slack channel found. This could occur when the partner slack channel name does not match the partner name from andela api. As such, retrying automation would continue to fail if the partner slack channel details are not updated manually

#### What are the relevant pivotal tracker stories?
[#165856629](https://www.pivotaltracker.com/story/show/165856629)

#### Postman Documentation
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/e66a6a83f30dca810bed)
